### PR TITLE
Add constants for hide and show cursor

### PR DESCRIPTION
**Pull request by iacore**

### CODE REVIEW

File: include/qubes-gui-protocol.h [hunk 1]
```diff
@@ -44,9 +44,11 @@ typedef unsigned __int32 uint32_t;
 #define MAX_WINDOW_HEIGHT 6144
 
 //cursor IDs
-#define CURSOR_DEFAULT      0
-#define CURSOR_X11      0x100  // use CURSOR_X11 + XC_* (from X11/cursorfont.h)
-#define CURSOR_X11_MAX  0x19a  // CURSOR_X11 + XC_num_glyphs
+#define CURSOR_DEFAULT       0
+#define CURSOR_X11      0x0100  // use CURSOR_X11 + XC_* (from X11/cursorfont.h)
+#define CURSOR_X11_MAX  0x019a  // CURSOR_X11 + XC_num_glyphs
+#define CURSOR_SHOW     0x1000
```
Comment (marmarek) [inline include/qubes-gui-protocol.h:50]: Is it necessary? `CURSOR_DEFAULT` (or some other specific one) should show the cursor, no?

File: include/qubes-gui-protocol.h [hunk 2]
```diff
@@ -44,9 +44,11 @@ typedef unsigned __int32 uint32_t;
...
+#define CURSOR_HIDE     0x2000
```
Comment (marmarek) [inline include/qubes-gui-protocol.h:51]: The whole 0x0000 - 0x00ff range is free for generic identifiers, you could use that as well (0x0001?).

**Review by marmarek — CHANGES_REQUESTED** [review]

File: include/qubes-gui-protocol.h (hunk 1 above)
Comment (iacore) [reply include/qubes-gui-protocol.h:50]: Show and hide cursors (xfixes) are independent from setting the cursor shape (x11, no extension)

File: include/qubes-gui-protocol.h (hunk 2 above)
Comment (iacore) [reply include/qubes-gui-protocol.h:51]: technically yes, but I want to reserve the last byte for actual cursor shapes, and second last byte for flags; we have 4 bytes to use

**Comment by iacore** [comment]
Closed for being impossible to implement. See https://github.com/QubesOS/qubes-issues/issues/2676.

**Pull request closed without merging**